internal/pkg/skuba/kubeadm/configmap.go

```diff
@@ -114,3 +117,46 @@ func RemoveAPIEndpointFromConfigMap(client clientset.Interface, node *v1.Node) e
 	}
 	return nil
 }
+
+// UpdateClusterConfigurationWithClusterVersion allows us to set certain configurations during init, but also during upgrades.
+// The configuration that we put here will be consistently set to newly created configurations, and when we upgrade a cluster.
+func UpdateClusterConfigurationWithClusterVersion(initConfiguration *kubeadmapi.InitConfiguration, version *version.Version) error {
+	if err := setApiserverAdmissionPlugins(initConfiguration); err != nil {
+		return err
+	}
+	if err := setContainerImagesWithClusterVersion(initConfiguration, version); err != nil {
+		return err
+	}
+	return nil
+}
+
+func setApiserverAdmissionPlugins(initConfiguration *kubeadmapi.InitConfiguration) error {
+	if initConfiguration.APIServer.ControlPlaneComponent.ExtraArgs == nil {
+		initConfiguration.APIServer.ControlPlaneComponent.ExtraArgs = map[string]string{}
+	}
+	admissionPlugins := []string{}
+	if len(initConfiguration.APIServer.ExtraArgs["enable-admission-plugins"]) > 0 {
+		admissionPlugins = strings.Split(initConfiguration.APIServer.ExtraArgs["enable-admission-plugins"], ",")
+	}
+	// List of recommended plugins: https://git.io/JemEu
+	admissionPlugins = append(admissionPlugins,
+		"NamespaceLifecycle",
+		"LimitRanger",
+		"ServiceAccount",
+		"TaintNodesByCondition",
+		"Priority",
+		"DefaultTolerationSeconds",
+		"DefaultStorageClass",
+		"PersistentVolumeClaimResize",
+		"MutatingAdmissionWebhook",
+		"ValidatingAdmissionWebhook",
+		"ResourceQuota",
+	)
```

Comment:
https://github.com/kubernetes/website/blob/master/content/en/docs/reference/access-authn-authz/admission-controllers.md#which-plugins-are-enabled-by-default

Any reason missing StorageObjectInUseProtection and RuntimeClass?

Reply:
Thank you, added them as a separate commit with an explanation of both.